### PR TITLE
os/bluestore: add blkin traces to BlueStore

### PR DESCRIPTION
**Pull request by yanghonggang**
Signed-off-by: Yang Honggang <yanghonggang@kuaishou.com>

### CODE REVIEW

**Comment by yanghonggang** [comment]
@xiexingguo hello xingguo，please help to review

**Comment by yanghonggang** [comment]
@ifed01  please help to review

**Comment by ifed01** [comment]
@yanghonggang - generally this looks good but I've never used lttng in Ceph begor. And now I'm having some troubles in making it work properly to verify your patch. Could you please help?
I'm following this instruction at (apart from using manually installed blkin):
https://github.com/ceph/ceph/blob/master/doc/dev/blkin.rst 
Building passes fine and I'm running vstart with osd_blkin_trace_all set to true. May be I should enable something else?

Then run rados bench: 
bin/rados bench -p foo 60 write -b 4K -t 16 --no-cleanup --write-omap
Once done, trying to inspect lttng reports. 
Unfortunately "lttng --list" shows nothing but:
UST events:
-------------
None

Starting LTTng session up as per the reference:
lttng create blkin-test
lttng enable-event --userspace zipkin:timestamp
lttng enable-event --userspace zipkin:keyval_integer
lttng enable-event --userspace zipkin:keyval_string
lttng start

doesn't help too. After stopping lttng and executing 'lttng view' I'm getting some outdated errors only:
Trace directory: /root/lttng-traces/blkin-test-20200612-153723

[error] Cannot open any trace for reading.

[error] opening trace "/root/lttng-traces/blkin-test-20200612-153723" for reading.

[error] none of the specified trace paths could be opened.

Timestamps above refer to the first ever lttng start attempt - nothing about later runs is reported.

Could you please advise how to make this stuff working?

**Comment by yanghonggang** [comment]
@ifed01 Maybe you forgot to start the lttng-sessiond daemon:
`lttng-sessiond -d --no-kernel`

**Comment by yanghonggang** [comment]
ping @ifed01

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -1545,6 +1549,15 @@ class BlueStore : public ObjectStore,
     }
 #endif
 
+    inline void set_state(state_t s) {
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.h:1552]: Suggest to make state private. And provide 'getter' to read it. Just to be absolutely sure that 'state' is properly encapsulated and accessed.

**Comment by ifed01** [comment]
@yanghonggang - looks like some relevant build errors...

**Comment by yanghonggang** [comment]
jenkins test make check

**Comment by yanghonggang** [comment]
> @yanghonggang - looks like some relevant build errors...

@ifed01 It is ok now

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-06-23_07:49:50-rados-wip-kefu-testing-2020-06-23-1007-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/

---

- http://pulpito.ceph.com/kchai-2020-06-23_07:49:50-rados-wip-kefu-testing-2020-06-23-1007-distro-basic-smithi/5171960, no clues, osd.2 failed to reply queries from osd.0 when peering. but there was no clues in its log.
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/5173648:  tracked by https://tracker.ceph.com/issues/45721
- http://pulpito.ceph.com/kchai-2020-06-23_15:15:59-rados-wip-kefu-testing-2020-06-23-1353-distro-basic-smithi/5173654/: no tests failed..

**Pull request merged**